src/shared/extension-release.c

```diff
@@ -8,108 +8,111 @@
-                if (extension_sysext_scope) {
-                        extension_sysext_scope_list = strv_split(extension_sysext_scope, WHITESPACE);
-                        if (!extension_sysext_scope_list)
+                scope = strv_env_pairs_get(image_release_file, confext_or_sysext_scope);
+                if (scope) {
+                        scope_list = strv_split(scope, WHITESPACE);
+                        if (!scope_list)
                                 return -ENOMEM;
                 }
 
-                /* by default extension are good for attachment in portable service and on the system */
+                /* By default extension are good for attachment in portable service and on the system */
                 valid = strv_contains(
-                                extension_sysext_scope_list ?: STRV_MAKE("system", "portable"),
-                                host_sysext_scope);
+                        scope_list ?: STRV_MAKE("system", "portable"),
+                        host_scope);
                 if (!valid) {
-                        log_debug("Extension '%s' is not suitable for scope %s, ignoring extension.", name, host_sysext_scope);
+                        log_debug("Extension '%s' is not suitable for scope %s, ignoring.", name, host_scope);
                         return 0;
                 }
         }
 
         /* When the architecture field is present and not '_any' it must match the host - for now just look at uname but in
          * the future we could check if the kernel also supports 32 bit or binfmt has a translator set up for the architecture */
-        extension_architecture = strv_env_pairs_get(extension_release, "ARCHITECTURE");
-        if (!isempty(extension_architecture) && !streq(extension_architecture, "_any") &&
-            !streq(architecture_to_string(uname_architecture()), extension_architecture)) {
+        confext_or_sysext_architecture = strv_env_pairs_get(image_release_file, "ARCHITECTURE");
+        if (!isempty(confext_or_sysext_architecture) && !streq(confext_or_sysext_architecture, "_any") &&
+        !streq(architecture_to_string(uname_architecture()), confext_or_sysext_architecture)) {
                 log_debug("Extension '%s' is for architecture '%s', but deployed on top of '%s'.",
-                          name, extension_architecture, architecture_to_string(uname_architecture()));
+                        name, confext_or_sysext_architecture, architecture_to_string(uname_architecture()));
                 return 0;
         }
 
-        extension_release_id = strv_env_pairs_get(extension_release, "ID");
-        if (isempty(extension_release_id)) {
-                log_debug("Extension '%s' does not contain ID in extension-release but requested to match '%s' or be '_any'",
-                          name, host_os_release_id);
+        confext_or_sysext_release_id = strv_env_pairs_get(image_release_file, "ID");
```

Comment:
we want to use "extension" as a generic term referring to both types, so you can leave most of these variables and names as they are, when they are used for either type